### ISSUE TITLE
Looks like godot is making a mess with file handling

### ISSUE THREAD

**Issue by VictorAssunc** [tetafro/godot#32]
## Context
I'm using https://github.com/valyala/quicktemplate to generate pages. `quicktemplate` generate code following the `.qtpl` file. So, if I have a template named `foo.qtpl` it generates a `foo.qtpl.go`.

## The problem
I have a 943-line `foo.qtpl` file that generates an 822-line `foo.qtpl.go` file. When I run `godot` for the `foo.qtpl.go` file, it fails with the following message: `Failed to run linter on file '/path/to/foo.qtpl.go': parse input file: inconsistency between file and AST: /path/to/foo.qtpl`.
**Pay attention to the file paths on the message.**

Debugging a little, I found where the error occurs: 
https://github.com/tetafro/godot/blob/a43fe64572ca20c576cc2ae017b442416eef9f89/getters.go#L63-L65
When printing `fmt.Println(len(pf.lines), p.Line)` I got the following results: `822 943`. Note that 822 is the number of lines that `foo.qtpl.go` has and that 943 is the number of lines in `foo.qtpl`. I didn't debug anymore, so I'm not sure if it's a `godot` or `golang` bug.

I can investigate further if I have time, but I don't promise.

Edit: I'm using the last godot version

**Comment by tetafro** [comment]
Yep, interesting. This is a kind of error that "should never happen" :)
My only guess without looking at the code, that quicktemplate produces code which is not formatted. And this is somehow affects godot. You can check it by running `gofmt -d /path/to/foo.qtpl.go`.

But of course it'd be easier if you just shared a reproducable example so I could debug it myself.

**Comment by igorscandido** [comment]
> Yep, interesting. This is a kind of error that "should never happen" :) My only guess without looking at the code, that quicktemplate produces code which is not formatted. And this is somehow affects godot. You can check it by running `gofmt -d /path/to/foo.qtpl.go`.
> 
> But of course it'd be easier if you just shared a reproducable example so I could debug it myself.

Hello @tetafro, I'm facing the same problem and I generated some code to try to help you debug it: https://github.com/igorscandido/GodotIssue-32

You can run `make godot` and you will see the same problem as @VictorAssunc described above.

It seems that the problem is that `.qtpl` file is bigger than `.qtpl.go` file, which causes it to fail.

For you to be able to teste both cases I separated a code that works (`.qtpl` size is smaller than `.qtpl.go`) in branch `success` and another that fails (with the problem me and @VictorAssunc described above) in the branch `error`.

**Comment by tetafro** [comment]
Thanks for the detailed report, it helped a lot.
TIL about the `line` compiler directive ([docs](https://gcc.gnu.org/onlinedocs/gccgo/Compiler-Directives.html)), which affects parsed code and sets [filenames](https://github.com/golang/go/blob/f7c5cbb82087c55aa82081e931e0142783700ce8/src/go/token/position.go#L22C17-L22C17). It was the source of the problem.

I fixed this in v1.4.16.
